trade-finance-manager-api/src/v1/middleware/validate-put-deal-cancellation-payload.ts

```diff
@@ -0,0 +1,9 @@
+import z from 'zod';
+import { TfmDealCancellation, createValidationMiddlewareForSchema } from '@ukef/dtfs2-common';
+import { DEAL_CANCELLATION } from '@ukef/dtfs2-common/schemas';
+
+const PutDealCancellationSchema: z.ZodType<Partial<TfmDealCancellation>> = DEAL_CANCELLATION.partial();
```

Comment:
Why does this have `partial`? I'm not familiar with Zod but I imagine we want to validate the whole schema and not part of it?

Reply:
This is just updating the cancellation request so its possible to fill in the `reason` on its own & the other fields be blank